packages/stex-react-renderer/src/lib/CompetencyIndicator.tsx

```diff
@@ -0,0 +1,147 @@
+import { getSectionInfo } from '@stex-react/utils';
+import { useContext, useEffect, useState } from 'react';
+import { DimIcon, ServerLinksContext } from './stex-react-renderer';
+import {
+  BloomDimension,
+  getDefiniedaInDoc,
+  getUriWeights,
+} from '@stex-react/api';
+import { DefiniendaItem, NumericCognitiveValues } from '@stex-react/api';
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  DialogContentText,
+} from '@mui/material';
+import LinearProgress from '@mui/material/LinearProgress';
+import CompetencyTable from './CompetencyTable';
+
+const CompetencyIndicator = ({ contentUrl }: { contentUrl: string }) => {
+  const [competencyData, setCompetencyData] = useState<
+    NumericCognitiveValues[] | null
+  >(null);
+  const [openDialog, setOpenDialog] = useState(false);
+  const [definedData, setDefinedData] = useState<DefiniendaItem[] | null>(null);
+  const { archive, filepath } = getSectionInfo(contentUrl);
+  const { mmtUrl } = useContext(ServerLinksContext);
+  const [URIs, setURIs] = useState<string[]>([]);
+
+  const handleOpenDialog = () => {
+    setOpenDialog(true);
+  };
+
+  const handleCloseDialog = () => {
+    setOpenDialog(false);
+  };
+
+  useEffect(() => {
+    async function fetchData() {
+      const data = await getDefiniedaInDoc(mmtUrl, archive, filepath);
+      console.log('defined data - ' + data);
+      setDefinedData(data);
+    }
+    fetchData();
+  }, [archive, filepath, mmtUrl]);
+
+  async function getData() {
```

Comment:
1. Better Name
2. Limit this function's scope by moving into useEffect.
3. You can altogether avoid writing async functions for such simple cases by using promises

```
useEffect(()=>{
      if (!definedData ) return;
      const URIs = definedData.flatMap((data) => data.symbols);
      setURIs(URIs);
      getUriWeights(URIs).then(data=>setCompetencyData(data));
}, [definedData])
```